cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -150,8 +161,12 @@ public InMemoryConfigurator(CConfiguration cConf, PluginFinder pluginFinder, Imp
     try (
       PluginInstantiator pluginInstantiator = new PluginInstantiator(cConf, app.getClass().getClassLoader(), tempDir)
     ) {
-      configurer = new DefaultAppConfigurer(appNamespace, artifactId, app,
-                                            configString, pluginFinder, pluginInstantiator);
+
+      configurer = new DefaultAppConfigurer(
+        appNamespace, artifactId, app, configString, pluginFinder, pluginInstantiator,
+        isRuntime ? new DefaultAppRuntimeConfigurer(appNamespace.getId(),
```

Comment:
nit: make this more readable by creating a runtime configurer property with the logic to select which one to use.